### PR TITLE
pid-plugin: Add wrappers for sched_* functions

### LINKED ISSUE #199: Virtualize PIDs for `sched_*` calls in the PID plugin
Currently, the PID plugin does not virtualize the PIDs for `sched_*` calls. We need to add wrappers for the `sched_*` calls that refer to the PIDs. The ability to restore things like CPU affinity can probably be delegated to a separate plugin. I'll create a separate issue for that.

### PR DESCRIPTION
**Pull request by rohgarg**
Fixes #199.

### CODE REVIEW

File: src/plugin/pid/sched_wrappers.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+#include "dmtcp.h"
+
+#define _real_sched_setaffinity     NEXT_FNC(sched_setaffinity)
+#define _real_sched_getaffinity     NEXT_FNC(sched_getaffinity)
+#define _real_sched_setscheduler    NEXT_FNC(sched_setscheduler)
+#define _real_sched_getscheduler    NEXT_FNC(sched_getscheduler)
+#define _real_sched_setattr         NEXT_FNC(sched_setattr)
+#define _real_sched_getattr         NEXT_FNC(sched_getattr)
+#define _real_sched_setparam        NEXT_FNC(sched_setparam)
+#define _real_sched_getparam        NEXT_FNC(sched_getparam)
```
Comment (karya0) [inline src/plugin/pid/sched_wrappers.cpp]: Can we instead follow the pattern from pidwrappers.h/pid_syscallsreal.c? This plugin is weird in that sense. Until we have a chance to do some cleanup, let's try to keep it consistent! :-)

**Comment by karya0** [comment]
LGTM modulo the _real_ stuff :)

**Comment by rohgarg** [comment]
@karya0, @gc00 : Could you please re-review the final patch? I believe it's ready to be merged.

**Comment by gc00** [comment]
It looks good to me.

For my own curiosity, @rohgarg, do you know why the Travis system was choking on compiling your earlier implementation of the `_real_` stuff?  Maybe there are some lessons for me, too.  :-)

**Comment by rohgarg** [comment]
@gc00: There were two separate issues:
 - There are two functions -- `sched_getattr()` and `sched_setattr()` -- that the man page says are present since kernel ver. 3.14. The man page also seems to imply that the function declarations and a `struct sched_attr` that these functions reference are present in a libc public header file: `sched.h`. And so, I added wrappers for these two, but turns out that neither of the two things I mentioned before are  exported publicly by libc.

  My fix is to ifdef out the two function wrappers; we can enable them in the future, if there is a requirement.

 - The second problem was with the test that I had added to the make-check test suite. The container environment Travis runs, apparently, doesn't support calls to setaffinity. My test, however, would assert and exit when it couldn't change the affinity of a running process. Note that I this wasn't a problem on other systems I tested.

  My fix is to comment out the asserts in the test program.

File: src/plugin/pid/pid_syscallsreal.c [hunk 2]
```diff
@@ -411,3 +412,52 @@ ssize_t _real_readlink(const char *path, char *buf, size_t bufsiz) {
...
+  REAL_FUNC_PASSTHROUGH(sched_setparam) (pid);
+}
+
+LIB_PRIVATE
+int _real_sched_getparam(pid_t pid, struct sched_param *param)
+{
+  REAL_FUNC_PASSTHROUGH(sched_getparam) (pid, param);
+}
+
+#if 0
```
Comment (karya0) [inline src/plugin/pid/pid_syscallsreal.c]: Can we instead add a configure check for these functions?

**Pull request merged**